syncthingwidgets/settings/settingsdialog.cpp

```diff
@@ -809,13 +810,19 @@ bool setAutostartEnabled(bool enabled)
         if (!desktopFile.open(QFile::WriteOnly | QFile::Truncate)) {
             return false;
         }
+#ifndef SYNCTHINGWIDGETS_AUTOSTART_EXEC_PATH
+        std::filesystem::path p = QCoreApplication::applicationFilePath().toStdString();
+        // Use the syncthingtray executable located in the directory of the applicationFilePath.
+        // Not using applicationFilePath directly to accomodate for NixOS as well, see:
+        // https://github.com/Martchus/syncthingtray/issues/176
+        std::string execPath = p.remove_filename().generic_string() + "syncthingtray";
+#else
+        std::string execPath = SYNCTHINGWIDGETS_AUTOSTART_EXEC_PATH;
```

Comment:
It makes no sense to instantiate this as `std::string` just to convert it to a `QString` afterwards. Please minimize the number of conversions.

Reply:
I agree, with https://doc.qt.io/qt-6/qcoreapplication.html#applicationDirPath this is not needed :)